src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -252,7 +252,7 @@ private bool VersionsFilter(object o)
             if ((UserInput.StartsWith("(", StringComparison.OrdinalIgnoreCase) || UserInput.StartsWith("[", StringComparison.OrdinalIgnoreCase)) &&
                VersionRange.TryParse(UserInput, out VersionRange userRange))
             {
-                if (o != null && NuGetVersion.TryParse(o.ToString(), out NuGetVersion userVersion))
+                if (o != null && NuGetVersion.TryParse(version.Version.ToString(), out NuGetVersion userVersion))
```

Comment:
`version.Version` is already a `NuGetVersion`: https://github.com/NuGet/NuGet.Client/blob/4f976ea2b3f32926ed585db4b21f6773762892b7/src/NuGet.Clients/NuGet.PackageManagement.UI/DisplayVersion.cs#L115

There's no need to serialize it into a string and then parse it again.